Traceback (most recent call last):
  File "/home/ubuntu/miniconda3/envs/prepa/lib/python3.9/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/home/ubuntu/miniconda3/envs/prepa/lib/python3.9/site-packages/nbclient/client.py", line 1112, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/home/ubuntu/miniconda3/envs/prepa/lib/python3.9/site-packages/nbclient/util.py", line 74, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/home/ubuntu/miniconda3/envs/prepa/lib/python3.9/site-packages/nbclient/util.py", line 53, in just_run
    return loop.run_until_complete(coro)
  File "/home/ubuntu/miniconda3/envs/prepa/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/home/ubuntu/miniconda3/envs/prepa/lib/python3.9/site-packages/nbclient/client.py", line 553, in async_execute
    await self.async_execute_cell(
  File "/home/ubuntu/miniconda3/envs/prepa/lib/python3.9/site-packages/nbclient/client.py", line 857, in async_execute_cell
    self._check_raise_for_error(cell, exec_reply)
  File "/home/ubuntu/miniconda3/envs/prepa/lib/python3.9/site-packages/nbclient/client.py", line 760, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
def estim_ud(u, i, E0, RP):
  """Fonction qui renvoie une estimation de Ud en calculant la moyenne des tensions pour lesquelles
  i est sup√©rieure ) E0/RP
  u  : vecteur numpy contenant les valeurs de tension
  i  : vecteur numpy contenant les valeurs d'intensit√©
  E0 : Valeur de E0 (flottant)
  RP : Valeur de RP (flottant)
  """
  imin = 0  # LIGNE A MODIFIER

  u_verif = []  # Liste qui contiendra les valeurs accept√©es.
  N = 0  # LIGNE A MODIFIER
  for j in range(N):
    if i[j] >= imin:
      u.append([])  # LIGNE A MODIFIER

  u_verif = np.array(u_verif)  # Transformation en vecteur numpy pour calculer la moyenne simplement.
  ud = np.mean(u_verif)
  return ud

"""Ecrire ci-dessous l'instruction qui va enregistrer dans la variable ud_estim l'estimation de ud
avec la m√©thode pr√©cdente"""
ud_estim = 0  # LIGNE A MODIFIER


"""Ajout du mod√®le sur le graphique.
"""
f,ax = plt.subplots()
f.suptitle("Mettre un titre")
ax.set_xlabel("L√©gender les X")
ax.set_ylabel("L√©gender les Y")
ax.errorbar(u, i, xerr=uu, yerr=ui, legend="L√©gender la courbe", linestyle="", color="black")
ax.plot([-5, ud_estim, ud_estim], [0, 0, 5]legend="L√©gender la courbe", linestyle="-",color="red")
ax.grid()
ax.legend()
plt.show()


"""La commande ci-apr√®s permettra de sauvegarder le graphique dans un fichier image
(dans le r√©pertoire o√π se trouve le notebook). Vous pourrez l'ins√©rer dans votre compte-rendu.
"""
f.savefig("caracteristique_diode.png")

------------------

[0;36m  File [0;32m"<ipython-input-2-6e16eb0175aa>"[0;36m, line [0;32m33[0m
[0;31m    ax.plot([-5, ud_estim, ud_estim], [0, 0, 5]legend="L√©gender la courbe", linestyle="-",color="red")[0m
[0m                                               ^[0m
[0;31mSyntaxError[0m[0;31m:[0m invalid syntax

SyntaxError: invalid syntax (<ipython-input-2-6e16eb0175aa>, line 33)

